www/apps/book/app/advanced-development/modules/isolation/page.mdx

```diff
@@ -21,12 +21,91 @@ For example, your custom module can't resolve the Product Module's main service
 
 ---
 
-## How to Implement Custom Features Across Modules?
+## Why are Modules Isolated
 
-In your Medusa application, you want to implement features that span across modules, or you want to extend an existing module's features and customize them for your own use case.
+Some of the module isolation's benefits include:
 
-For example, you want to extend the Product Module to add new properties to the `Product` data model.
+- Integrate your module into any Medusa application without side-effects to your setup.
+- Replace existing modules with your custom implementation, if your use case is drastically different.
+- Use modules in other environments, such as Edge functions and Next.js apps.
 
-Medusa provides the tools to implement these use cases while maintaining isolation between modules.
+---
+
+## How to Extend Data Model of Another Module?
+
+To extend the data model of another module, such as the `product` data model of the Product Module, use Medusa's linking concepts as explained in the [Module Links chapters](../../module-links/page.mdx).
+
+---
+
+## How to Use Services of Other Modules?
+
+If you're building a feature that uses functionalities from different modules, use a workflow whose steps resolve the modules' services to perform these functionalities.
+
+Workflows ensure data consistency through their roll-back mechanism and tracking of each execution's status, steps, input, and output.
+
+### Example
+
+For example, consider you have two modules:
+
+1. A module that stores and manages brands in your application.
+2. A module that integrates a third-party Content Management System (CMS).
+
+To sync brands from your application to the third-party system, create the following steps:
+
+export const stepsHighlights = [
+  ["1", "retrieveBrandsStep", "A step that retrieves brands using a brand module."],
+  ["14", "createBrandsInCmsStep", "A step that creates brands using a CMS module."],
+  ["25", "", "Add a compensation function to the step if an error occurs."]
+]
+
+```ts title="Example Steps" highlights={stepsHighlights}
+const retrieveBrandsStep = createStep(
+  "retrieve-brands",
+  async (_, { container }) => {
+    const brandModuleService = container.resolve(
+      "brandModuleService"
+    )
+
+    const brands = await brandModuleService.listBrands()
+
+    return new StepResponse(brands)
+  }
+)
```

Comment:
nit(non blocking): should we show the useRemoteQueryStep?

Reply:
In this case I want to showcase the usage of the different modules, so I think it's fine to not use `useRemoteQueryStep`.